src/sentry/services/hybrid_cloud/user/impl.py

```diff
@@ -42,29 +42,30 @@ def serialize_many(
     def get_many(self, *, filter: UserFilterArgs) -> List[RpcUser]:
         return self._FQ.get_many(filter)
 
+    def get_many_ids(self, *, filter: UserFilterArgs) -> List[int]:
+        return self._FQ.get_many_ids(filter)
+
     def get_many_by_email(
         self,
         emails: List[str],
         is_active: bool = True,
         is_verified: bool = True,
-        is_project_member: bool = False,
-        project_id: Optional[int] = None,
+        organization_id: Optional[int] = None,
     ) -> List[RpcUser]:
         query = self._FQ.base_query()
         if is_verified:
             query = query.filter(emails__is_verified=is_verified)
         if is_active:
             query = query.filter(is_active=is_active)
-        if is_project_member:
-            query = query.filter(
-                sentry_orgmember_set__organizationmemberteam__team__projectteam__project_id__in=[
-                    project_id
-                ]
-            )
-        return [
+        if organization_id is not None:
+            query = query.filter(orgmembermapping_set__organization_id=organization_id)
+
+        users = [
             self._FQ.serialize_rpc(user)
             for user in query.filter(in_iexact("emails__email", emails))
         ]
+        users_by_emails = {email: user for user in users for email in user.emails}
```

Comment:
Indexing users by email is risky. We have no guarantees around uniqueness of emails. It is possible that multiple users in an organization could have the same email address, and have that email address verified.

Reply:
Good point, let me expand this a bit then.

Reply:
Got a fix!